### PR TITLE
Vim plugins: added new and updated

### PR DESCRIPTION
**Pull request by jupblb**
###### Motivation for this change
Added the following plugins:
- [nvim-neoclip.lua](https://github.com/AckslD/nvim-neoclip.lua)
- [nvim-jqx](https://github.com/gennaro-tedesco/nvim-jqx)
- [nvim-metals](https://github.com/scalameta/nvim-metals)
- [telescope-lsp-handlers.nvim](https://github.com/gbrlsnchs/telescope-lsp-handlers.nvim)
- [telescope-vim-bookmarks.nvim](https://github.com/tom-anders/telescope-vim-bookmarks.nvim)
- [venn.nvim](https://github.com/jbyuki/venn.nvim)

Also performed an overall update. Since some of the added plugins have an irrelevant Makefile, some overrides with `dontBuild = true` were implemented.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -661,6 +665,10 @@ self: super: {
     };
   });
 
+  venn-nvim = super.venn-nvim.overrideAttrs (old: {
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:668]: There is no Makefile in the repo https://github.com/jbyuki/venn.nvim/

Comment (jupblb) [reply pkgs/misc/vim-plugins/overrides.nix]: Thanks! Good point. Fixed. :)

**Comment by teto** [comment]
rebase please

**Comment by jonringer** [comment]
remove the update commit, and it hsould be fine

**Comment by jupblb** [comment]
> remove the update commit, and it should be fine

Done

**Pull request merged**